packager/certcache/certcache.go

```diff
@@ -318,6 +324,39 @@ func (this *CertCache) isHealthy(ocspResp []byte) error {
 	return nil
 }
 
+func (this *CertCache) readOCSPHelper(numTries int, exhaustedRetries bool) ([]byte, time.Time, error) {
+	var ocspUpdateAfter time.Time
+
+	this.certsMu.RLock()
+	defer this.certsMu.RUnlock()
+	ocsp, err := this.ocspFile.Read(context.Background(), this.shouldUpdateOCSP, func(orig []byte) []byte {
+		return this.fetchOCSP(orig, this.certs, &ocspUpdateAfter, numTries > 0)
+	})
+	if err != nil {
+		if exhaustedRetries {
+			return nil, time.Time{}, errors.Wrap(err, "Updating OCSP cache")
+		} else {
+			return nil, time.Time{}, nil
+		}
+	}
+	if len(ocsp) == 0 {
+		if exhaustedRetries {
+			return nil, time.Time{}, errors.New("Missing OCSP response.")
+		} else {
+			return nil, time.Time{}, nil
+		}
+	}
+	if err := this.isHealthy(ocsp); err != nil {
+		if exhaustedRetries {
+			return nil, time.Time{}, errors.Wrap(err, "OCSP failed health check.")
+		} else {
+			return nil, time.Time{}, nil
+		}
+	}
+
+	return ocsp, ocspUpdateAfter, nil
+}
+
 // Returns the OCSP response and expiry, refreshing if necessary.
 // TODO(banaag): Per twifkak's suggestion, consider:
 // It may also be interesting to try to separate the retry logic from the fetch logic. One approach comes to mind:
```

Comment:
I think you just did this TODO, essentially?

Reply:
Will address in follow-up PR #369